packages/vue-app/template/App.js

```diff
@@ -296,19 +297,23 @@ export default {
     <% } /* features.layouts */ %>
     <% if (isFullStatic) { %>
     getRouterBase() {
-      return (this.$router.options.base || '').replace(/\/+$/, '')
+      return withoutTrailingSlash(this.$router.options.base)
     },
     getRoutePath(route = '/') {
       const base = this.getRouterBase()
       if (base && route.startsWith(base)) {
         route = route.substr(base.length)
       }
-      return (route.replace(/\/+$/, '') || '/').split('?')[0].split('#')[0]
+      let path = route.split('?')[0].split('#')[0]
+      <% if (!nuxtOptions.router.trailingSlash) { %>
+        path = withoutTrailingSlash(path)
+      <% } %>
```

Comment:
We could normalise to no trailing slashes, which would also solve this problem, but would have the downside of breaking the distinction between parent/child routes (see #8423 - thoughts welcome @pi0).